cli/tools/info.rs

```diff
@@ -121,564 +128,456 @@ fn print_cache_info(
   }
 }
 
-const SIBLING_CONNECTOR: char = '├';
-const LAST_SIBLING_CONNECTOR: char = '└';
-const CHILD_DEPS_CONNECTOR: char = '┬';
-const CHILD_NO_DEPS_CONNECTOR: char = '─';
-const VERTICAL_CONNECTOR: char = '│';
-const EMPTY_CONNECTOR: char = ' ';
-
-fn fmt_module_graph(graph: &ModuleGraph, f: &mut impl Write) -> fmt::Result {
-  if graph.roots.is_empty() || graph.roots.len() > 1 {
-    return writeln!(
-      f,
-      "{} displaying graphs that have multiple roots is not supported.",
-      colors::red("error:")
-    );
+struct TreeNode {
+  text: String,
+  children: Vec<TreeNode>,
+}
+
+impl TreeNode {
+  pub fn from_text(text: String) -> Self {
+    Self {
+      text,
+      children: Default::default(),
+    }
   }
-  let root_specifier = graph.resolve(&graph.roots[0].0);
-  match graph.try_get(&root_specifier) {
-    Ok(Some(root)) => {
-      if let Some(cache_info) = root.maybe_cache_info.as_ref() {
-        if let Some(local) = &cache_info.local {
-          writeln!(
-            f,
-            "{} {}",
-            colors::bold("local:"),
-            local.to_string_lossy()
-          )?;
-        }
-        if let Some(emit) = &cache_info.emit {
-          writeln!(f, "{} {}", colors::bold("emit:"), emit.to_string_lossy())?;
-        }
-        if let Some(map) = &cache_info.map {
-          writeln!(f, "{} {}", colors::bold("map:"), map.to_string_lossy())?;
-        }
-      }
-      writeln!(f, "{} {}", colors::bold("type:"), root.media_type)?;
-      let modules = graph.modules();
-      let total_size: f64 = modules.iter().map(|m| m.size() as f64).sum();
-      let dep_count = modules.len() - 1;
-      writeln!(
-        f,
-        "{} {} unique {}",
-        colors::bold("dependencies:"),
-        dep_count,
-        colors::gray(format!("(total {})", display::human_size(total_size)))
-      )?;
+}
+
+fn print_tree_node<TWrite: Write>(
+  tree_node: &TreeNode,
+  writer: &mut TWrite,
+) -> fmt::Result {
+  fn print_children<TWrite: Write>(
+    writer: &mut TWrite,
+    prefix: &str,
+    children: &Vec<TreeNode>,
+  ) -> fmt::Result {
+    const SIBLING_CONNECTOR: char = '├';
+    const LAST_SIBLING_CONNECTOR: char = '└';
+    const CHILD_DEPS_CONNECTOR: char = '┬';
+    const CHILD_NO_DEPS_CONNECTOR: char = '─';
+    const VERTICAL_CONNECTOR: char = '│';
+    const EMPTY_CONNECTOR: char = ' ';
+
+    let child_len = children.len();
+    for (index, child) in children.iter().enumerate() {
+      let is_last = index + 1 == child_len;
+      let sibling_connector = if is_last {
+        LAST_SIBLING_CONNECTOR
+      } else {
+        SIBLING_CONNECTOR
+      };
+      let child_connector = if child.children.is_empty() {
+        CHILD_NO_DEPS_CONNECTOR
+      } else {
+        CHILD_DEPS_CONNECTOR
+      };
       writeln!(
```

Comment:
Suggest to wrap this in `write_to_stdout_ignore_sigpipe`

Reply:
It does that at a higher level. This is writing to a string.